src/spurt/utils/_tiling.py

```diff
@@ -0,0 +1,314 @@
+from __future__ import annotations
+
+from typing import Protocol, runtime_checkable
+
+import numpy as np
+from numpy.typing import ArrayLike
+
+
+def intersects(box1: ArrayLike, box2: ArrayLike) -> bool:
+    """Check if two boxes intersect."""
+    return not (
+        box1[2] < box2[0] or box1[0] > box2[2] or box1[1] > box2[3] or box1[3] < box2[1]
+    )
+
+
+@runtime_checkable
+class TilerInterface(Protocol):
+    """Rectangular tiler interface."""
+
+    @property
+    def ntiles(self) -> int:
+        """Number of tiles."""
+
+    @property
+    def bounds(self) -> tuple[int, int]:
+        """Shape of rectangle being tiled."""
+
+    @property
+    def tiles(self) -> ArrayLike:
+        """2D array of shape (ntiles, 4) using shapely bbox convention."""
+
+    @property
+    def neighbors(self) -> ArrayLike | None:
+        """2D array of shape (nedges, 2).
+
+        This represents sorted pair of indices of neighboring tiles similar to
+        edges in a graph.
+        """
+
+
+class RegularTiler(TilerInterface):
+    """Regularly sized tiles."""
+
+    def __init__(self, shape: tuple[int, int], max_tiles: int, dilation: float = 0.05):
+        """Regularly sized tileset.
+
+        Parameters
+        ----------
+        shape: tuple[int, int]
+            Shape of rectangle to tile up.
+        max_tiles: int
+            Maximum number of tiles. Use of perfect squares recommended.
```

Comment:
Adding to Geoff's comment- If we are doing `tiles_per_dim = int(np.sqrt(self._max_tiles))`, maybe something like this would be more clear 
```suggestion
            Maximum number of tiles.
            Actual number of tiles is the perfect square `<= max_tiles`.
```


```python
In [39]: [(n_in, spurt.utils.RegularTiler((1234, 2345), n_in).ntiles) for n_in in range(1, 11)]
Out[39]:
[(1, 1),
 (2, 1),
 (3, 1),
 (4, 4),
 (5, 4),
 (6, 4),
 (7, 4),
 (8, 4),
 (9, 9),
 (10, 9)]
```

Reply:
Done